plugin-process-to-cgroup-bridge/src/transform.rs

```diff
@@ -0,0 +1,199 @@
+        for measurement in measurements.iter() {
+            if let Some(cgroup_measurement) = self.cgroup_measurement_from_measurement(measurement)? {
+                cgroup_measurements.push(cgroup_measurement);
+                if self.keep_processed_measurements {
+                    kept_measurements.push(measurement.clone());
+                }
+            } else {
+                kept_measurements.push(measurement.clone());
+            }
+        }
+        measurements.clear();
+
+        if self.merge_similar_cgroups {
+            let mut aggregated_cgroup_measurements = Self::aggregate_cgroups_measurements(&mut cgroup_measurements);
+            cgroup_measurements.clear();
+            cgroup_measurements.merge(&mut aggregated_cgroup_measurements);
+        }
+
+        measurements.merge(&mut kept_measurements);
+        measurements.merge(&mut cgroup_measurements);
+
+        Ok(())
+    }
+}
+
+impl ProcessToCgroupBridgeTransform {
+    pub fn new(
+        processes_metrics_ids: Vec<RawMetricId>,
+        merge_similar_cgroups: bool,
+        keep_processed_measurements: bool,
+    ) -> Self {
+        Self {
+            processes_metrics_ids,
+            merge_similar_cgroups,
+            keep_processed_measurements,
+        }
+    }
+
+    fn cgroup_measurement_from_measurement(
+        &self,
+        measurement: &MeasurementPoint,
+    ) -> anyhow::Result<Option<MeasurementPoint>> {
+        if !self.processes_metrics_ids.contains(&measurement.metric) {
+            return Ok(None);
+        }
+        let pid = match Self::extract_process_id_from_measurement(measurement) {
+            Ok(pid) => pid,
+            Err(_) => return Ok(None),
+        };
+        match Self::find_cgroup_path_from_process_id(pid)? {
+            Some(cgroup_path) => {
+                let cgroup_consumer = ResourceConsumer::ControlGroup {
+                    path: cgroup_path.into(),
+                };
+                let mut cgroup_measurement = measurement.clone();
+                cgroup_measurement.consumer = cgroup_consumer;
+                Ok(Some(cgroup_measurement))
+            }
+            None => {
+                log::warn!("no cgroup was found for process {pid}");
+                Ok(None)
+            }
+        }
+    }
+
+    fn find_cgroup_path_from_process_id(pid: u32) -> anyhow::Result<Option<String>> {
+        let procfs_cgroup_base_path = get_proc_path();
+
+        let procfs_cgroup_filepath = procfs_cgroup_base_path.join(pid.to_string()).join("cgroup");
+
+        if !Path::new(&procfs_cgroup_filepath).exists() {
+            return Ok(None);
+        }
+
+        let contents = fs::read_to_string(&procfs_cgroup_filepath)
+            .with_context(|| format!("Failed to read {:?}", procfs_cgroup_filepath))?;
+
+        // a typical procfs cgroup file will contain only one line
+        // eg: 0::/system.slice/docker-7c7fc86f5f2a609c41c6edd65bd1b64135124a687fa6516f6b177b040d6e3b68.scope
```

Comment:
This is only true for cgroup v2. For cgroup v1, it contains one line per cgroup hierarchy.
Example:
```
2:memory:/daemons
5:cpuacct,cpu:/daemons
```